docker/src/main/java/org/jclouds/docker/domain/NetworkSettings.java

```diff
@@ -98,13 +180,84 @@ public Builder ports(Map<String, List<Map<String, String>>> ports) {
+         this.linkLocalIPv6Address = linkLocalIPv6Address;
+         return this;
+      }
+
+      public Builder linkLocalIPv6PrefixLen(int linkLocalIPv6PrefixLen) {
+         this.linkLocalIPv6PrefixLen = linkLocalIPv6PrefixLen;
+         return this;
+      }
+
+      public Builder sandboxKey(String sandboxKey) {
+         this.sandboxKey = sandboxKey;
+         return this;
+      }
+
+      public Builder secondaryIPAddresses(List<String> secondaryIPAddresses) {
+         this.secondaryIPAddresses = secondaryIPAddresses;
+         return this;
+      }
+
+      public Builder secondaryIPv6Addresses(List<String> secondaryIPv6Addresses) {
+         this.secondaryIPv6Addresses = secondaryIPv6Addresses;
+         return this;
+      }
+
+      public Builder endpointId(String endpointId) {
+         this.endpointId = endpointId;
+         return this;
+      }
+
+      public Builder globalIPv6Address(String globalIPv6Address) {
+         this.globalIPv6Address = globalIPv6Address;
+         return this;
+      }
+
+      public Builder globalIPv6PrefixLen(int globalIPv6PrefixLen) {
+         this.globalIPv6PrefixLen = globalIPv6PrefixLen;
+         return this;
+      }
+
+      public Builder ipv6Gateway(String ipv6Gateway) {
+         this.ipv6Gateway = ipv6Gateway;
+         return this;
+      }
+
+      public Builder macAddress(String macAddress) {
+         this.macAddress = macAddress;
+         return this;
+      }
+
+      public Builder networks(Map<String, Details> networks) {
+         this.networks.putAll(networks);
+         return this;
+      }
+
       public NetworkSettings build() {
-         return NetworkSettings.create(ipAddress, ipPrefixLen, gateway, bridge, portMapping, ports);
+         return NetworkSettings.create(bridge, sandboxId, hairpinMode, linkLocalIPv6Address, linkLocalIPv6PrefixLen, ports,
+                 sandboxKey, secondaryIPAddresses, secondaryIPv6Addresses, endpointId, gateway,
+                 globalIPv6Address, globalIPv6PrefixLen, ipAddress, ipPrefixLen, ipv6Gateway, macAddress, networks, portMapping);
       }
 
       public Builder fromNetworkSettings(NetworkSettings in) {
          return this.ipAddress(in.ipAddress()).ipPrefixLen(in.ipPrefixLen()).gateway(in.gateway()).bridge(in.bridge())
-               .portMapping(in.portMapping()).ports(in.ports());
+               .portMapping(in.portMapping()).ports(in.ports()).sandboxId(in.sandboxId()).hairpinMode(in.hairpinMode()).linkLocalIPv6Address(in
+                         .linkLocalIPv6Address()).linkLocalIPv6PrefixLen(in.linkLocalIPv6PrefixLen()).sandboxKey(in.sandboxKey()).secondaryIPAddresses(in
+                         .secondaryIPAddresses()).secondaryIPv6Addresses(in.secondaryIPv6Addresses()).endpointId(in.endpointId()).globalIPv6Address(in
+                         .globalIPv6Address()).globalIPv6PrefixLen(in.globalIPv6PrefixLen()).ipv6Gateway(in.ipv6Gateway()).macAddress(in.macAddress())
+                 .networks(in.networks());
```

Comment:
Not for this PR, but we should consider moving all builders to google auto builders.